File: parse_postman.dart
import 'dart:convert';
import 'dart:io';

void extractEndpoints(List itemList, String prefix) {
  for (var item in itemList) {
    if (item.containsKey('item')) {
      var folderName = item['name'] ?? 'Unknown Folder';
      extractEndpoints(item['item'], prefix + folderName + ' / ');
    } else if (item.containsKey('request')) {
      var req = item['request'];
      var method = req['method'] ?? 'UNKNOWN';
      var url = '';
      if (req['url'] is Map) {
        url = req['url']['raw'] ?? '';
      } else if (req['url'] is String) {
        url = req['url'];
      }
      var name = item['name'] ?? 'Unnamed Request';
      print('$prefix$name [$method] $url');
    }
  }
}

void main() async {
  var filePath = 'D:/AppPro/tugas15flutter/ABSENSI PPKD B3.postman_collection.json';
  try {
    var file = File(filePath);
    var content = await file.readAsString();
    var data = jsonDecode(content);
    
    var info = data['info'] ?? {};
    print('Collection Name: ${info['name'] ?? 'Unknown'}');
    
    var items = data['item'] ?? [];
    extractEndpoints(items, '');
  } catch (e) {
    print('Error: $e');
  }
}

File: parse_postman.py
import json
import sys

def extract_endpoints(item_list, prefix=""):
    endpoints = []
    for item in item_list:
        if "item" in item:
            # It's a folder
            folder_name = item.get("name", "Unknown Folder")
            endpoints.extend(extract_endpoints(item["item"], prefix + folder_name + " / "))
        elif "request" in item:
            # It's a request
            req = item["request"]
            method = req.get("method", "UNKNOWN")
            url = ""
            if isinstance(req.get("url"), dict):
                url = req["url"].get("raw", "")
            elif isinstance(req.get("url"), str):
                url = req["url"]
            name = item.get("name", "Unnamed Request")
            endpoints.append(f"{prefix}{name} [{method}] {url}")
    return endpoints

if __name__ == "__main__":
    file_path = "D:/AppPro/tugas15flutter/ABSENSI PPKD B3.postman_collection.json"
    try:
        with open(file_path, "r", encoding="utf-8") as f:
            data = json.load(f)
        
        print(f"Collection Name: {data.get('info', {}).get('name', 'Unknown')}")
        endpoints = extract_endpoints(data.get("item", []))
        for ep in endpoints:
            print(ep)
    except Exception as e:
        print(f"Error: {e}")
